core/src/main/java/org/mule/runtime/core/util/ClassUtils.java

```diff
@@ -1208,4 +1171,55 @@ public static boolean isConsumable(Class<?> payloadClass)
         }
         return false;
     }
-}
+
+    /**
+     * Executes the given {@code supplier} using the given {@code classLoader}
+     * as the current {@link Thread}'s context classloader.
+     * <p>
+     * This method guarantees that whatever the outcome, the thread's context classloader
+     * is set back to the value that it had before executing this method
+     *
+     * @param classLoader the context {@link ClassLoader} on which the {@code runnable} should be executed
+     * @param supplier    a {@link Supplier}
+     * @return the value that the {@code supplier} produced
+     */
+    public static <T> T withClassLoader(ClassLoader classLoader, Supplier<T> supplier)
+    {
+        final Thread currentThread = Thread.currentThread();
+        final ClassLoader currentClassLoader = currentThread.getContextClassLoader();
+        currentThread.setContextClassLoader(classLoader);
+        try
+        {
+            return supplier.get();
+        }
+        finally
+        {
+            currentThread.setContextClassLoader(currentClassLoader);
+        }
+    }
+
+    /**
+     * Executes the given {@code runnable} using the given {@code classLoader}
+     * as the current {@link Thread}'s context classloader.
+     * <p>
+     * This method guarantees that whatever the outcome, the thread's context classloader
+     * is set back to the value that it had before executing this method
+     *
+     * @param classLoader the context {@link ClassLoader} on which the {@code runnable} should be executed
+     * @param runnable    a closure
+     */
+    public static void withClassLoader(ClassLoader classLoader, Runnable runnable)
+    {
+        final Thread currentThread = Thread.currentThread();
```

Comment:
Runnable and Supplier are not replaceable with each other. I would have to wrap the runnable into a supplier which returns null. IMO just as ugly as the current solution